gapil/resolver/api.go

```diff
@@ -164,7 +164,8 @@ func resolve(rv *resolver) {
 		removeDeadCode(rv, f.Block)
 	}
 	for _, f := range rv.api.Functions {
-		resolveFenceOrder(rv, f, visitedFuncs{})
+		recursive := false
+		resolveFenceOrder(rv, f, visitedFuncs{}, false, &recursive)
```

Comment:
Maybe just add `Recursive` field to `semantic.Function` ?

Reply:
Done